### PR TITLE
docs: adding contribution guidelines

### PR DESCRIPTION
**Pull request by mike-hoang**
Signed-off-by: Michael Hoang <mhoang@redhat.com>

### What does this PR do?:
Adds a contributing file which is a requirement for CNCF onboarding

### Which issue(s) this PR fixes:
https://github.com/devfile/api/issues/772

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
>  - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
>  - test/doc updates are made as part of this PR
> -  If unchecked, explain why it's not needed

- [ ] Unit/Functional tests

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

- [x] Documentation 

- [ ] Client Impact

### CODE REVIEW

**Review by johnmcollier — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+## Code of Conduct
+
+Before contributing to this repository for the first time, please review our project's [Code of Conduct](https://github.com/devfile/api/blob/main/CODE_OF_CONDUCT.md).
+
+## Getting Started
+
+### Issues
+
+- Issues are tracked via the the [devfile/api](https://github.com/devfile/api) repo. Open or search for issues on the [devfile/api](https://github.com/devfile/api) repo with the label `area/api`.
```
Comment (yangcao77) [inline CONTRIBUTING.md:13]:
```suggestion
- Open or search for [issues](https://github.com/devfile/api/issues) with the label `area/api`.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+- If a related issue doesn't exist, you can open a new issue using a relevant [issue form](https://github.com/devfile/api/issues/new/choose). You can tag Devfile Registry related issues with the `/area api` text in your issue.
+
+- Search for existing pull requests or issues that are related before you start a PR.
```
Comment (yangcao77) [inline CONTRIBUTING.md:17]: duplicated with first point

**Review by yangcao77 — CHANGES_REQUESTED** [review]
Left some comments. 
Can you also add a guide for testing? and also a guide for building?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+`Signed-off-by: First Lastname <email@email.com>`
+
+Once you set your user.name and user.email in your git config, you can sign your commit automatically with git commit -s. When you think the code is ready for review, create a pull request and link the issue associated with it.
+
+Owners of the repository will watch out for and review new PRs.
+
+If comments have been given in a review, they have to be addressed before merging.
+
+After addressing review comments, don’t forget to add a comment in the PR afterward, so everyone gets notified by Github and know to re-review.
```
Comment (amisevsk) [inline CONTRIBUTING.md:31]:
```suggestion
After addressing review comments, don’t forget to add a comment in the PR afterward, so everyone gets notified by Github and knows to re-review.
```

**Review by amisevsk — COMMENTED** [review]
Generally looks good to me, apart from Stephanie's review.

File: CONTRIBUTING.md (hunk 2 above)
Comment (mike-hoang) [reply CONTRIBUTING.md]: removed

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+- If a related issue doesn't exist, you can open a new issue using a relevant [issue form](https://github.com/devfile/api/issues/new/choose). You can tag Devfile Registry related issues with the `/area api` text in your issue.
+
+- Search for existing pull requests or issues that are related before you start a PR.
+
+### Pull Requests
+
+Note: All commits must be signed off with the footer:
+
+`Signed-off-by: First Lastname <email@email.com>`
```
[the remark this answers is not in the document]
Comment (mike-hoang) [reply CONTRIBUTING.md]: removed the inline and switched to block

**Review by yangcao77 — APPROVED** [review]
changes lgtm

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+- If a related issue doesn't exist, you can open a new issue using a relevant [issue form](https://github.com/devfile/api/issues/new/choose). You can tag issues with `/area api`.
+
+### Building
+
+To build the CRD and the various schemas, you don't need to install any pre-requisite apart from `docker`.
+In the root directory, just run the following command:
+
+```
+./docker-run.sh ./build.sh
+```
```
Comment (amisevsk) [inline CONTRIBUTING.md:24]: Worth noting (perhaps) that on some SELinux configurations, this command fails "permission denied". The current workaround I use is to add `:Z` to the volume mount, though this is probably not ideal.

**Review by amisevsk — APPROVED** [review]

File: CONTRIBUTING.md (hunk 5 above)
Comment (amisevsk) [reply CONTRIBUTING.md:24]: :+1: makes sense to me

**Pull request merged**